framework/test/unit/modules/chain/unit/modules/blocks/process.js

```diff
@@ -1661,200 +1666,182 @@ describe('blocks/process', async () => {
 	});
 
 	describe('generateBlock', async () => {
-		describe('modules.transactions.getUnconfirmedTransactionList', async () => {
-			describe('when query returns empty array', async () => {
-				beforeEach(() => {
+		describe('modules.transactions.getUnconfirmedTransactionList', () => {
+			describe('when query returns empty array', () => {
+				beforeEach(async () => {
 					modules.transactions.getUnconfirmedTransactionList.returns([]);
-					return modules.blocks.verify.processBlock.callsArgWith(
+					modules.processTransactions.verifyTransactions.returns(
+						Promise.resolve([])
+					);
+					modules.blocks.verify.processBlock.callsArgWith(
 						3,
 						null,
 						modules.blocks.verify.processBlock.args
 					);
 				});
 
-				it('should generate block without transactions', done => {
+				it('should generate block without transactions', async () => {
 					blocksProcessModule.generateBlock(
 						{ publicKey: '123abc', privateKey: 'aaa' },
 						41287231,
 						err => {
 							expect(err).to.be.null;
-							expect(library.logic.transaction.verify.calledOnce).to.be.false;
 							expect(
 								modules.blocks.verify.processBlock.args[0][0].transactions
 									.length
 							).to.equal(0);
-							done();
 						}
 					);
 				});
 			});
 
-			describe('when query returns undefined', async () => {
-				beforeEach(() => {
+			describe('when query returns undefined', () => {
+				beforeEach(async () => {
 					modules.transactions.getUnconfirmedTransactionList.returns(undefined);
-					return modules.blocks.verify.processBlock.callsArgWith(
+					modules.processTransactions.verifyTransactions.returns(
+						Promise.resolve([])
+					);
+					modules.blocks.verify.processBlock.callsArgWith(
 						3,
 						null,
 						modules.blocks.verify.processBlock.args
 					);
 				});
 
-				it('should generate block without transactions', done => {
+				it('should generate block without transactions', async () => {
 					blocksProcessModule.generateBlock(
 						{ publicKey: '123abc', privateKey: 'aaa' },
 						41287231,
 						err => {
 							expect(err).to.be.null;
-							expect(library.logic.transaction.verify.calledOnce).to.be.false;
 							expect(
 								modules.blocks.verify.processBlock.args[0][0].transactions
 									.length
 							).to.equal(0);
-							done();
 						}
 					);
 				});
 			});
 
-			describe('when query returns transactions', async () => {
-				beforeEach(() => {
+			describe('when query returns transactions', () => {
+				beforeEach(async () => {
 					modules.transactions.getUnconfirmedTransactionList.returns([
-						{ id: 1, type: 0 },
-						{ id: 2, type: 1 },
+						{ id: 1, type: 0, senderPublicKey: '1L' },
```

Comment:
this is `senderId` =0
and why do we need to add new property here?

Reply:
I added `senderPublicKey` so it's easy to differentiate from a `transaction response` object which looks very similar. Really all that's actually needed for this test is `id`. I can change to `senderId` or just have `id` also.